src/features/user/controllers/user-controller.test.ts

```diff
@@ -9,54 +9,55 @@ import { UserFindByIdService } from '@/features/user/services/user-find-by-id-se
 import { HttpError } from '@/shared/errors/http-error';
 import { UserNotFound } from '@/shared/errors/user-not-found-error';
 import { HttpStatusCode } from '@/shared/protocols/http-client';
+import { bcryptAdapteMock } from '@/shared/test-helpers/mocks/bcryptAdapter.mock';
 import { accountRepositoryMock } from '@/shared/test-helpers/mocks/repositories/account-repository.mock';
 import { userRepositoryMock } from '@/shared/test-helpers/mocks/repositories/user-repository.mock';
 
-const makeSut = () => {
-  const mockUserCreateService = mock<UserCreateService>(
-    new UserCreateService(userRepositoryMock, {
-      compare: vi.fn(),
-      encrypt: vi.fn(),
-    })
-  );
-  const mockUserFindByIdService = mock<UserFindByIdService>(
-    new UserFindByIdService(userRepositoryMock)
-  );
-
-  const mockGetUserAccountsService = mock<GetUserAccountsService>(
-    new GetUserAccountsService(userRepositoryMock, accountRepositoryMock)
-  );
-
-  const userController = new UserController(
-    mockUserCreateService,
-    mockUserFindByIdService,
-    mockGetUserAccountsService
-  );
-
-  const req = mockDeep<Request>();
-  const res = {
-    json: vi.fn(),
-    status: vi.fn().mockReturnThis(),
-  } as unknown as Response;
-  const next = vi.fn();
-
-  return {
-    getUserAccountsService: mockGetUserAccountsService,
-    next,
-    req,
-    res,
-    userController,
-    userCreateService: mockUserCreateService,
-    userFindByIdService: mockUserFindByIdService,
-  };
-};
-
 describe('[Controllers] UserController', () => {
+  let mockUserCreateService: UserCreateService;
+  let mockUserFindByIdService: UserFindByIdService;
+  let mockGetUserAccountsService: GetUserAccountsService;
+  let userController: UserController;
+
+  let req: Request;
+  let res: Response;
+  let next: NextFunction;
+  let error: HttpError;
+
+  beforeEach(() => {
+    mockUserCreateService = mock<UserCreateService>(
+      new UserCreateService(userRepositoryMock, bcryptAdapteMock)
+    );
+
+    mockUserFindByIdService = mock<UserFindByIdService>(
+      new UserFindByIdService(userRepositoryMock)
+    );
+
+    mockGetUserAccountsService = mock<GetUserAccountsService>(
+      new GetUserAccountsService(userRepositoryMock, accountRepositoryMock)
+    );
+
+    userController = new UserController(
+      mockUserCreateService,
+      mockUserFindByIdService,
+      mockGetUserAccountsService
+    );
+
+    req = mockDeep<Request>();
+
+    res = {
+      json: vi.fn(),
+      status: vi.fn().mockReturnThis(),
+    } as unknown as Response;
+
+    next = vi.fn() as unknown as NextFunction;
```

Comment:
aqui podemos melhorar essa linha
```suggestion
    next: NextFunction = vi.fn();
```

Reply:
Não da por conta do mock, ele fica chamadno o router e tive que fazer ele virar unknown e depois nextfuntion